lib/ParseSIL/ParseSIL.cpp

```diff
@@ -5394,9 +5394,15 @@ bool SILParserTUState::parseDeclSILStage(Parser &P) {
 static Optional<VarDecl *> lookupGlobalDecl(Identifier GlobalName,
                                             SILLinkage GlobalLinkage,
                                             SILType GlobalType, Parser &P) {
+  // Create a set of DemangleOptions to produce the global variable's
+  // identifier, which is used as a search key in the declaration context.
+  Demangle::DemangleOptions demangleOpts;
+  demangleOpts.QualifyEntities = false;
+  demangleOpts.ShowPrivateDiscriminators = false;
+  demangleOpts.DisplayEntityTypes = false;
```

Comment:
@jrose-apple do these demangle options look reasonable?

Reply:
Yeah, that looks fine. QualifyEntities does seem to make a number of the other options irrelevant.